docker-app/qfieldcloud/core/tests/test_project.py

```diff
@@ -330,3 +336,53 @@ def test_public_projecs(self):
         self.assertEqual(json[1]["owner"], "user2")
         self.assertEqual(json[1]["user_role"], "reader")
         self.assertEqual(json[1]["user_role_origin"], "public")
+
+    def test_private_project_memberships(self):
+        """Tests for QF-1553 - limit collaboration on private projects"""
+
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token1.key)
+
+        # Create a project with a collaborator
+        u = User.objects.create(username="u")
+        o = Organization.objects.create(username="o", organization_owner=u)
+        p = Project.objects.create(name="p", owner=u, is_public=True)
+
+        apiurl = f"/api/v1/projects/{p.pk}/"
+
+        self.client.raise_request_exception = True
+
+        def assert_no_role():
+            response = self.client.get(apiurl, follow=True)
+            self.assertEqual(response.status_code, 403)
+
+        def assert_role(user_role, user_role_origin):
+            response = self.client.get(apiurl, follow=True)
+            self.assertEqual(response.status_code, 200)
+            json = response.json()
+            self.assertEqual(json["user_role"], user_role)
+            self.assertEqual(json["user_role_origin"], user_role_origin)
+
+        # Project is public, we have a public role
+        # TODO: this fails, is is expected the queryset does not contain public projects ?
+        # TODO: if so, it should 404 (or 403) instead of 500
+        # assert_role(user_role="reader", user_role_origin="public")
```

Comment:
@suricactus to review

Reply:
I would say we should not include public projects, yes. Or at least we should have a step to filter them. Returning 404 sounds good to me.